Sources/BraveWallet/Crypto/Portfolio/PortfolioView.swift

```diff
@@ -213,6 +176,174 @@ struct PortfolioView: View {
+              assetSectionsHeader
+            }
+          }
+        )
+      }
+    }
+  }
+  
+  /// Builds the list of assets without any grouping or expandable / collapse behaviour.
+  @ViewBuilder private func ungroupedAssets(with group: AssetGroupViewModel) -> some View {
+    ForEach(group.assets) { asset in
+      Button(action: {
+        selectedToken = asset.token
+      }) {
+        PortfolioAssetView(
+          image: AssetIconView(
+            token: asset.token,
+            network: asset.network,
+            shouldShowNetworkIcon: true
+          ),
+          title: asset.token.name,
+          symbol: asset.token.symbol,
+          networkName: asset.network.chainName,
+          amount: asset.fiatAmount(currencyFormatter: portfolioStore.currencyFormatter),
+          quantity: asset.quantity
+        )
+      }
+    }
+  }
+  
+  @State var groupToggleState: [AssetGroupViewModel.ID: Bool] = [:]
+  
+  /// Builds the expandable/collapseable (expanded by default) section content for a given group.
+  @ViewBuilder private func groupedAssetSections(with group: AssetGroupViewModel) -> some View {
+    DisclosureGroup(
+      isExpanded: Binding(
+        get: { groupToggleState[group.id, default: true] },
+        set: { isExpanded in
+          groupToggleState[group.id] = isExpanded
+        }
+      ),
+      content: {
+        ForEach(group.assets) { asset in
+          Button(action: {
+            selectedToken = asset.token
+          }) {
+            PortfolioAssetView(
+              image: AssetIconView(
+                token: asset.token,
+                network: asset.network,
+                shouldShowNetworkIcon: true
+              ),
+              title: asset.token.name,
+              symbol: asset.token.symbol,
+              networkName: asset.network.chainName,
+              amount: asset.fiatAmount(currencyFormatter: portfolioStore.currencyFormatter),
+              quantity: asset.quantity
+            )
+          }
+        }
+        .listRowInsets(.init(top: 0, leading: 0, bottom: 0, trailing: 16))
+      },
+      label: {
+        if case let .account(account) = group.groupType {
+          AddressView(address: account.address) {
+            groupHeader(for: group)
+          }
+        } else {
+          groupHeader(for: group)
+        }
+      }
+    )
+  }
+  
+  /// Builds the in-section header for an AssetGroupViewModel that is shown in expanded and non-expanded state. Not used for ungrouped assets.
+  private func groupHeader(for group: AssetGroupViewModel) -> some View {
+    VStack(spacing: 0) {
+      HStack {
+        if case let .network(networkInfo) = group.groupType {
+          NetworkIcon(network: networkInfo, length: 32)
+        } else if case let .account(accountInfo) = group.groupType {
+          Blockie(address: accountInfo.address, shape: .rectangle)
+            .frame(width: 32, height: 32)
+            .clipShape(RoundedRectangle(cornerRadius: 4))
```

Comment:
i think we should aligned the account blockie shape through out the wallet.

Reply:
The new v2 designs for other views might have square blockies as well but I don't think we should be updating the entire wallet in this Portfolio Grouping PR.